lib/elixir/lib/enum.ex

```diff
@@ -1408,7 +1408,14 @@ defmodule Enum do
 
   """
   @spec sum(t) :: number
-  def sum(enumerable) do
+  def sum(first..last) when last > first do
+    ((last * (last + 1)) / 2) - ((first * (first - 1)) / 2)    
+  end
+  def sum(first..last) do
```

Comment:
I think we need a newline between function clauses to match previous style. It's still the same function.